pkg/loadbalancer/redirectpolicy/controller.go

```diff
@@ -322,13 +322,18 @@ func (c *lrpController) updateRedirects(wtxn writer.WriteTxn, ws *statedb.WatchS
 					}
 					continue
 				}
+				//if we only have one frontend mapping, we dont need the frontend port name so it will not check the port name in the backend ports
+				if len(lrp.FrontendMappings) == 1 {
+					feM.fePort = lb.FEPortName("")
+				}
```

Comment:
Why is this required?

Reply:
I found this is required because we have the test for this.

https://github.com/cilium/cilium/blob/main/cilium-cli/connectivity/builder/manifests/local-redirect-policy.yaml#L11-L19. Maybe the question is why we have such test where the frontend name is different from backend. If we already allow this in the past, I dont know if we need to keep it in case there is the surprise when people upgrade to 1.18? I can also change the test.

Reply:
Good spotting! Let's fix the test -- I don't remember any particular reason for the mismatched port names.

Reply:
hope this is not a feature that we offered to the users before.